internal/querycoordv2/observers/leader_observer.go

```diff
@@ -168,18 +168,15 @@ func (o *LeaderObserver) findNeedRemovedSegments(leaderView *meta.LeaderView, di
 	for _, s := range dists {
 		distMap[s.GetID()] = struct{}{}
 	}
+	// if a segment(segment_id, node_id) only exists in the leader view, it must be cleaned up
 	for sid, s := range leaderView.Segments {
 		_, ok := distMap[sid]
-		existInCurrentTarget := o.target.GetHistoricalSegment(leaderView.CollectionID, sid, meta.CurrentTarget) != nil
-		existInNextTarget := o.target.GetHistoricalSegment(leaderView.CollectionID, sid, meta.NextTarget) != nil
-		if ok || existInCurrentTarget || existInNextTarget {
+		if ok {
 			continue
 		}
 		log.Debug("leader observer append a segment to remove:", zap.Int64("collectionID", leaderView.CollectionID),
```

Comment:
the log could be simplifed: "stale leader view segment found", the ":" symbol is not needed